projects/vdk-plugins/vdk-control-api-auth/src/vdk/plugin/control_api_auth/base_auth.py

```diff
@@ -0,0 +1,284 @@
+from vdk.plugin.control_api_auth.auth_config import InMemAuthConfiguration
+from vdk.plugin.control_api_auth.auth_exception import VDKAuthException
+from vdk.plugin.control_api_auth.auth_request_values import AuthRequestValues
+from vdk.plugin.control_api_auth.login_types import LoginTypes
+
+log = logging.getLogger(__name__)
+
+
+class AuthenticationCache:
+    """
+    Class used to keep the fields in the vdk-cred.json file from which
+    authorization and authentication tokens can be taken. Used only when
+    caching is enabled.
+    """
+
+    def __init__(
+        self,
+        authorization_url=None,
+        refresh_token=None,
+        access_token=None,
+        access_token_expiration_time=0,
+        client_id=None,
+        client_secret=None,
+        api_token=None,
+        api_token_authorization_url=None,
+        auth_type=None,
+        **ignored_kwargs,
+    ):
+        """
+        :param authorization_url:
+            Token or Authorization URI used to exchange grant for access token
+        :param refresh_token:
+            Used in refresh token grant type flow;
+            See https://datatracker.ietf.org/doc/html/rfc6749#section-1.5
+        :param access_token:
+            Cached access token used during requests;
+            See https://datatracker.ietf.org/doc/html/rfc6749#section-1.4
+        :param access_token_expiration_time:
+            When the access token expires
+            (epoch seconds -- same as time.time() + "expires_in" value returned
+            by the authorization server that generates the access token.)
+        :param client_id:
+            The client identifier;
+            See https://datatracker.ietf.org/doc/html/rfc6749#section-2.3.1
+        :param api_token:
+            Token used to grant access to the API,
+            including exchange for access token.
+        :param api_token_authorization_url:
+            The URL which exchanges api_token for access token.
+        """
+        #
+        self.authorization_url = authorization_url
+        self.refresh_token = refresh_token
+        self.access_token = access_token
+        self.access_token_expiration_time = access_token_expiration_time
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.api_token = api_token
+        self.api_token_authorization_url = api_token_authorization_url
+        self.auth_type = auth_type
+
+    def __eq__(self, other):
+        return self.__dict__ == other.__dict__
+
+
+class AuthenticationCacheSerDe:
+    """Class used for cache serialization and deserialization."""
+
+    @staticmethod
+    def serialize(cache) -> str:
+        return json.dumps(cache.__dict__)
+
+    @staticmethod
+    def deserialize(content: str) -> AuthenticationCache:
+        if content:
+            return AuthenticationCache(**json.loads(content))
+        else:
+            return AuthenticationCache()
+
+
+class BaseAuth:
```

Comment:
why `BaseAuth` in favour of the old `Authentication`? is it abstract and meant for extension? don't see any prior extensions;
also, transitioning the imports only is going to be easier

Reply:
There will be a different `Authentication` api with more generalized logic.